components/bounty-form.js

```diff
@@ -49,6 +51,33 @@ export function BountyForm ({
     `
   )
 
+  const submitUpsertBounty = useCallback(
+    // we ignore the invoice since only stackers can post bounties
+    // the invoice is only for funding the wallet
+    async (_, boost, bounty, values, __) => {
+      const { error } = await upsertBounty({
+        variables: {
+          sub: item?.subName || sub?.name,
+          id: item?.id,
+          boost: boost ? Number(boost) : undefined,
+          bounty: bounty ? Number(bounty) : undefined,
+          ...values
+        }
+      })
+      if (error) {
+        throw new Error({ message: error.toString() })
+      }
+
+      if (item) {
+        await router.push(`/items/${item.id}`)
+      } else {
+        const prefix = sub?.name ? `/~${sub.name}` : ''
+        await router.push(prefix + '/recent')
+      }
+    }, [upsertBounty, router])
+
+  const anonUpsertBounty = useAnonymous(submitUpsertBounty, { requireSession: true })
```

Comment:
Maybe `lib/anonymous.js` and all related nomenclature should be renamed the `useAnonymous` hook can now also be used to only handle invoices for stackers and specifically exclude anon users using `{ requireSession: true }` ... so it's more about "pay as you go" than enabling anon users now, I think.